### ISSUE TITLE
The WoWAnalyzer premium plan

### ISSUE THREAD

**Issue by MartijnHols** [WoWAnalyzer/WoWAnalyzer#1741]
*This is just announcing the plan, we're still a long way from implementing it.*

# The problem
This project takes a lot of time to work on, and motivation to continue working on things fluctuates. I'm not sure how much longer I'll continue spending loads of my time on this myself as my motivation is running out as well. I also want to start spending more time on projects that earn me money to a point where I can work less for an employer and more on my own projects.

Still there's a lot of work that has to be done continuously to continue helping users; primarily spec analysis needs to be updated, we need to adapt to changes in the game, combat log and the WCL API, and PRs need to be reviewed in a timeline manner (among other things). We also need to do a bunch of things to prepare for potential catastrophes, such as by building a combat log parser.

# The solution
While generally a poor motivator, money can provide that extra needed incentive to work on things. Making a couple hundred dollar as a student by doing what you like doing is a lot better than getting a job you dislike. There are also people spending several hours farming gold each month to get WoW tokens for their subscriptions ($15 per month) that could perhaps replace this time with developing something here for his or her spec.

I don't see any other possible solution that could be a solution to increase contributions to the project. Especially things like reviewing PRs is a painful process that is hard to incentivize.

To make this possible I am planning to introduce ads and a Premium model.

## Ads
Ads would be something like AdSense, showing a couple of non-intrusive banners on the results page and possibly homepage. I've already tried signing up with Google AdSense, but so far they're complaining about having insufficient content. I'm not sure if they just have issues with the SPA - we might have to end up with another platform.

I reckon there are a lot of ad blockers among gamers, we can ask them to unblock us but even that likely won't help. We could start blocking features that were funded by ads/premium if this turns out to be a big problem. I don't think this would be a nice thing to do, but at the same time it does a decent job at making it crystal clear that the site doesn't develop itself.

## Premium
<img align="right" src="https://user-images.githubusercontent.com/4565223/40587152-ad1d6798-61cb-11e8-97c6-015fec6e86dc.gif">
The premium model for starters would remove ads from the site, and further on lock (new) features behind it. At this time I don't have a clear idea of valid features to restrict behind premium, but something like the possibility of filtering by phases could be developed for premium only. **The primary reward for people will be getting (more) updates to the site and their specs.**

The plan right now is to offer two ways of getting premium:
- **Patreon**: at the low fee of $1 per month users get an ad-free experience and access to the premium features (this might still change).
- **GitHub**: any code merged to [`master`](https://github.com/WoWAnalyzer/WoWAnalyzer/commits/master) earns a user 30 days of premium (this might still change). There won't initially be a minimum requirement for contributions for premium via GitHub, I'm hoping that small contributions will lead to bigger changes at a later time.

This is aimed at increasing the amount of contributions to the site; both platforms lead to this.

## Distributing the income
I've been thinking for a long time about how to fairly distribute any money coming in. One of the possibilities has been to use a model where spec maintainers are paid based on the usage of their spec implementation and possibly a per review reward for reviewing PRs. I've been told Icy Veins uses a similar model where their guide writers [earn at least $500 per month](https://www.reddit.com/r/wow/comments/7vi27a/icy_veins_recruiting_class_guide_writers_paid/) depending on views and reviewers ~$30 per hour.

This solution has several challenges: several specs are maintained by multiple people, several people contributed parts of specs, how do we encourage working on parts of the site that affects all specs, how do we get people to continue developing things when their spec is already good enough, and most importantly how do we legally pay out everyone across countries. I'm based in The Netherlands and we have fairly strict regulations coming in for independent contractors that could conflict. The gist of it is that you can't earn less than minimum wage, which might still be the case since some features take a lot of time to develop. While it's uncertain this would apply to us, I don't really want to spend a lot of time investigating and talking to lawyers.

The solution I have pretty much decided on is to use Bountysource. With Bountysource we put bounties on [issues in GitHub](https://github.com/WoWAnalyzer/WoWAnalyzer/issues) for things that need to be developed, and whoever implements it gets awarded the full bounty. I've been experimenting with this for a while now and while it hasn't been very successful, I'm hopeful that an increase in the amounts will make it more popular.

With Bountysource the persons that contributed the work are rewarded so there's no complexity in determining who deserves the reward. It also only encourages future work and doesn't care about what is already there, this better aligns with the goal of encouraging more development.

The system can be applied to anything that we can describe in an issue, so it can work for non-spec specific things as well. It allows me to prioritize things as I think will help the most users, so even specs without currently active maintainers can be breathed new life into. And the best thing is that Bountysource takes care of and are responsible for the legality of things so I won't have to talk to any lawyers.

# In conclusion
This plan should lead to more contributions to the site which should make our analysis more extensive and more accurate. And with contributors also being rewarded everyone wins!

I'd love to hear what you think about this plan. Your feedback would be much appreciated!

TODO:

- [ ]  Ads
- [ ] Premium login: https://github.com/WoWAnalyzer/WoWAnalyzer/issues/437

**Comment by MartijnHols** [comment]
The plan for now is to show Patreon ads to reserve space for ads and worry about AdSense accepting us later. I was thinking ads and premium had to be released together for the best user reception, but I’m starting to consider that maybe we should add the “ads” first so instead users *want* an option to remove them and we provide that later. In any case at this point I’m not far from finishing so maybe that’s too late.

**Comment by MartijnHols** [comment]
Everything is ready and already running on https://bfa.wowanalyzer.com. I'll update the Patreon page and make a news article and then this should be good to launch. Getting Google AdSense to work can be done later, for now self-advertising will have to do.

**Comment by MartijnHols** [comment]
Launched just now!
